saleor/payment/gateways/np_atobarai/plugin.py

```diff
@@ -0,0 +1,164 @@
+        merchant_code=conf[MERCHANT_CODE],
+        sp_code=conf[SP_CODE],
+        terminal_id=conf[TERMINAL_ID],
+    )
+
+
+class NPAtobaraiGatewayPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.payments.np-atobarai"
+    PLUGIN_NAME = GATEWAY_NAME
+    CONFIGURATION_PER_CHANNEL = True
+    SUPPORTED_CURRENCIES = "JPY"
+
+    DEFAULT_CONFIGURATION = [
+        {"name": MERCHANT_CODE, "value": None},
+        {"name": SP_CODE, "value": None},
+        {"name": TERMINAL_ID, "value": None},
+        {"name": USE_SANDBOX, "value": True},
+    ]
+
+    CONFIG_STRUCTURE = {
+        MERCHANT_CODE: {
+            "type": ConfigurationTypeField.SECRET,
+            "help_text": "Provide NP後払い Merchant Code.",
+            "label": "Merchant Code",
+        },
+        SP_CODE: {
+            "type": ConfigurationTypeField.SECRET,
+            "help_text": "Provide NP後払い SP Code.",
+            "label": "SP Code",
+        },
+        TERMINAL_ID: {
+            "type": ConfigurationTypeField.SECRET,
+            "help_text": "Provide NP後払い Terminal ID.",
+            "label": "Terminal ID",
+        },
+        USE_SANDBOX: {
+            "type": ConfigurationTypeField.BOOLEAN,
+            "help_text": "Determines if Saleor should use NP後払い sandbox API.",
+            "label": "Use sandbox",
+        },
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        configuration = {item["name"]: item["value"] for item in self.configuration}
+        self.config = GatewayConfig(
+            gateway_name=GATEWAY_NAME,
+            auto_capture=False,
+            supported_currencies=self.SUPPORTED_CURRENCIES,
+            connection_params={
+                "merchant_code": configuration[MERCHANT_CODE],
+                "sp_code": configuration[SP_CODE],
+                "terminal_id": configuration[TERMINAL_ID],
+                "sandbox_mode": configuration[USE_SANDBOX],
+            },
+        )
+
+    def _get_gateway_config(self) -> GatewayConfig:
+        return self.config
+
+    @require_active_plugin
+    def capture_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return capture(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
+    def refund_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return refund(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
+    def void_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return void(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
+    def process_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return process_payment(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
```

Comment:
@korycins Is this decorator needed? I think we discussed that at some point the plugins manager took over the task of checking whether a plugin is active before calling its methods?

Reply:
Yes, I think this is already merged in 3.0 and master.